File: DKPlugins/DKRocket_OBSOLETE/DKRocketLocation.cpp
#ifdef USE_DKDuktape 
#include "DK/DKApp.h"
#include "DKRocket/DKRocketLocation.h"
#include "DKRocket/DKRocket.h"


///////////////////////
bool DKRocketLocation::Init()
{
	DKDEBUGFUNC();
	DKDuktape::AttachFunction("DKRocketLocation_hash", DKRocketLocation::hash);
	DKDuktape::AttachFunction("DKRocketLocation_host", DKRocketLocation::host);
	DKDuktape::AttachFunction("DKRocketLocation_hostname", DKRocketLocation::hostname);
	DKDuktape::AttachFunction("DKRocketLocation_href", DKRocketLocation::href);
	DKDuktape::AttachFunction("DKRocketLocation_origin", DKRocketLocation::origin);
	DKDuktape::AttachFunction("DKRocketLocation_password", DKRocketLocation::password);
	DKDuktape::AttachFunction("DKRocketLocation_pathname", DKRocketLocation::pathname);
	DKDuktape::AttachFunction("DKRocketLocation_port", DKRocketLocation::port);
	DKDuktape::AttachFunction("DKRocketLocation_protocol", DKRocketLocation::protocol);
	DKDuktape::AttachFunction("DKRocketLocation_search", DKRocketLocation::search);
	DKDuktape::AttachFunction("DKRocketLocation_username", DKRocketLocation::username);
	
	DKClass::DKCreate("DKRocket/DKRocketLocation.js");
	return true;
}

//////////////////////////////////////
int DKRocketLocation::hash(duk_context* ctx)
{
	int found = DKRocket::Get()->href.find("#");
	DKString hash;
	if(found > 0){
		hash = DKRocket::Get()->href.substr(found);
	}
	duk_push_string(ctx, hash.c_str());
	return true;
}

//////////////////////////////////////
int DKRocketLocation::host(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////////
int DKRocketLocation::hostname(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////
int DKRocketLocation::href(duk_context* ctx)
{
	if(duk_is_string(ctx, 0)){
		DKString url = duk_get_string(ctx, 0);
		DKRocket::Get()->LoadUrl(url);
	}
	duk_push_string(ctx, DKRocket::Get()->href.c_str());
	return true;
}

////////////////////////////////////////
int DKRocketLocation::origin(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////////
int DKRocketLocation::password(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////////
int DKRocketLocation::pathname(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////
int DKRocketLocation::port(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////////
int DKRocketLocation::protocol(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

////////////////////////////////////////
int DKRocketLocation::search(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

//////////////////////////////////////////
int DKRocketLocation::username(duk_context* ctx)
{
	DKString host = "TODO";
	duk_push_string(ctx, host.c_str());
	return true;
}

#endif //USE_DKDuktape

File: DKPlugins/DKRocket_OBSOLETE/DKRocketLocation.js
//https://developer.mozilla.org/en-US/docs/Web/API/Locationc

var location;

////////////////////////////////
var Location = function(pointer)
{
	Object.defineProperty(this, "hash", { get: function(){ return DKRocketLocation_hash(); } });
	Object.defineProperty(this, "host", {
		get: function()   { return DKRocketLocation_host();      },
		set: function(val){ return DKRocketLocation_host(val);   }
	});
	Object.defineProperty(this, "hostname", {
		get: function()   { return DKRocketLocation_hostname();      },
		set: function(val){ return DKRocketLocation_hostname(val);   }
	});
	Object.defineProperty(this, "href", {
		get: function()   { return DKRocketLocation_href();      },
		set: function(val){ return DKRocketLocation_href(val);   }
	});
	Object.defineProperty(this, "origin", { get: function(){ return DKRocketLocation_origin(); } });
	Object.defineProperty(this, "pathname", {
		get: function()   { return DKRocketLocation_pathname();      },
		set: function(val){ return DKRocketLocation_pathname(val);   }
	});
	Object.defineProperty(this, "port", {
		get: function()   { return DKRocketLocation_port();      },
		set: function(val){ return DKRocketLocation_port(val);   }
	});
	Object.defineProperty(this, "protocol", {
		get: function()   { return DKRocketLocation_protocol();      },
		set: function(val){ return DKRocketLocation_protocol(val);   }
	});
	Object.defineProperty(this, "search", {
		get: function()   { return DKRocketLocation_search();      },
		set: function(val){ return DKRocketLocation_search(val);   }
	});
	
	Location.prototype.assign = function(url){
		DKRocketLocation_href(url); 
	}
	Location.prototype.reload = function(forcedReload){
		DKRocketLocation_reload(forcedReload); 
	}
	Location.prototype.replace = function(url){
		DKRocketLocation_href(url); //TODO - https://developer.mozilla.org/en-US/docs/Web/API/Location/replace
	}
}

location = new Location("location");